felix/bpf/ut/bpf_prog_test.go

```diff
@@ -836,51 +842,91 @@ var ipv4Default = &layers.IPv4{
 	Protocol: layers.IPProtocolUDP,
 }
 
+var srcIPv6 = net.IP([]byte{0xff, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1})
+var dstIPv6 = net.IP([]byte{0xff, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2})
+
+var ipv6Default = &layers.IPv6{
+	Version:    6,
+	HopLimit:   64,
+	SrcIP:      srcIPv6,
+	DstIP:      dstIPv6,
+	NextHeader: layers.IPProtocolUDP,
+}
+
 var udpDefault = &layers.UDP{
 	SrcPort: 1234,
 	DstPort: 5678,
 }
 
-func testPacket(ethAlt *layers.Ethernet, ipv4Alt *layers.IPv4, l4Alt gopacket.Layer, payloadAlt []byte) (
-	*layers.Ethernet, *layers.IPv4, gopacket.Layer, []byte, []byte, error) {
+func testPacket(ethAlt *layers.Ethernet, ipv4Alt *layers.IPv4, ipv6Alt *layers.IPv6, l4Alt gopacket.Layer, payloadAlt []byte, ipv6Enabled bool) (
```

Comment:
if you did not add the extra ipv6 argument, but kept the `testPacket` as is and introduced a new `testPacketv6` and moved the body of this function to a new function like `testPacket46` (say) which would be internal and could take both v4 and v6 headers, then you would not need to touch any of the existing tests, would you?

Reply:
Right, fixed it. Yeah, that is much better:-)